formulaire-public/src/main/resources/public/ts/controllers/recap.ts

```diff
@@ -105,16 +112,22 @@ export const recapController = ng.controller('RecapController', ['$scope',
     };
 
     const getQuestionIdsFromPositionHistoric = () : number[] => {
-        let validatedElements = new FormElements();
-        validatedElements.all = vm.formElements.all.filter(e => vm.historicPosition.indexOf(e.position) >= 0);
-        return validatedElements.getAllQuestions().all.map(q => q.id);
+        let validatedElements: FormElements = new FormElements();
+        validatedElements.all = vm.formElements.all.filter((e: FormElement) => vm.historicPosition.indexOf(e.position) >= 0);
+        return validatedElements.getAllQuestionsAndChildren().all.map(q => q.id);
     }
 
     const checkMandatoryQuestions = async (validatedQuestionIds: number[]) : Promise<boolean> => {
-        let mandatoryQuestions = vm.formElements.getAllQuestions().all.filter(q => q.mandatory && validatedQuestionIds.indexOf(q.id) >= 0);
+        let mandatoryQuestions: Question[] = vm.formElements.getAllQuestions().all.filter((q: Question) => q.mandatory && validatedQuestionIds.indexOf(q.id) >= 0);
         for (let question of mandatoryQuestions) {
-            let responses = vm.responses.all.filter(r => r.question_id === question.id && r.answer);
-            if (responses.length <= 0) {
+            if (question.question_type === Types.MATRIX) {
+                for (let child of question.children.all) {
+                    if (vm.responses.all.filter((r: Response) => r.question_id === child.id && r.answer).length <= 0) {
+                        return false;
+                    }
+                }
+            }
+            else if (vm.responses.all.filter((r: Response) => r.question_id === question.id && r.answer).length <= 0) {
                 return false;
             }
```

Comment:
mieux encore
return !vm.responses.all.filter((r: Response) => question.children.all.find(child => r.question_id === child.id) && r.answer).length <= 0 || vm.responses.all.filter((r: Response) => r.question_id === question.id && r.answer).length <= 0

tu réduit ton for a 2 lignes après, tu peux gérer ta condition de retour comme tu veux (genre si tu veux seulement retourner la valeur quand c'est faux, etc